### DISCUSSION TITLE
Validation rule doesn't update to match props update

### DISCUSSION THREAD

**Discussion by calpage18** [react-hook-form/react-hook-form#1774] [General] [answered]
Hi,

I have a form for adding credit card details. By default, the card security code length is 3, but if the card being used is an American Express, it should be 4 digits. I'm using this validation rule:

`const validateInput = (value) => {
    return value.length === props.length;
  };`

It works fine when entering a 3 digit security code (the initial value of `props.length`), but when `props.length` changes to `4`, it's still validating against the original value of `3` from `props.length`.

Is there a way to get the validation callback to update and read the new value for `props.length`?

Thanks!

**Comment by calpage18** [comment, marked as answer]
Never mind, just read a different discussion about `unregister`!

**Reply by bluebill1049** [reply]
also take a look at this issue as well: https://github.com/react-hook-form/react-hook-form/issues/1749 for changing rules with controller
